tests/Game.class.test.js

```diff
@@ -0,0 +1,916 @@
+          const nonEmptyCells = initialColumn.filter((cell) => cell !== 0);
+          const cursor = nonEmptyCells[0];
+
+          expect(checkIsCorrectlyAligned(column, 'end', cursor)).toBe(true);
+
+          nonEmptyCells.forEach((cell) => {
+            expect(column.includes(cell)).toBe(true);
+          });
+        });
+      });
+
+      it('should merge cells with the same value down', () => {
+        const initialState = [
+          [16, 0, 0, 0],
+          [0, 0, 8, 8],
+          [0, 8, 8, 0],
+          [8, 8, 0, 16],
+        ];
+
+        const game2048 = new Game(cloneState(initialState));
+
+        game2048.start();
+        game2048.moveDown();
+
+        const state = game2048.getState();
+        const stateTransposed = transposeState(state);
+
+        stateTransposed.forEach((column) => {
+          expect(column.includes(16)).toBe(true);
+        });
+      });
+
+      it('should merge multiple cells with the same value down', () => {
+        const initialState = [
+          [8, 16, 32, 16],
+          [8, 16, 32, 16],
+          [16, 8, 8, 16],
+          [16, 8, 8, 16],
+        ];
+
+        const game2048 = new Game(cloneState(initialState));
+
+        game2048.start();
+        game2048.moveDown();
+
+        const getCell = getCellFromState(game2048);
+
+        expect(getCell(2, 0)).toBe(16);
+        expect(getCell(2, 1)).toBe(32);
+        expect(getCell(2, 2)).toBe(64);
+        expect(getCell(2, 3)).toBe(32);
+        expect(getCell(3, 0)).toBe(32);
+        expect(getCell(3, 1)).toBe(16);
+        expect(getCell(3, 2)).toBe(16);
+        expect(getCell(3, 3)).toBe(32);
+      });
+    });
+  });
+
+  describe('game status', () => {
+    it('should be a "playing" when board is full but moves are still possible', () => {
+      const game2048 = new Game([
+        [128, 128, 0, 8],
+        [16, 8, 16, 32],
+        [8, 16, 32, 64],
+        [16, 32, 0, 128],
+      ]);
+
+      game2048.start();
+
+      expect(game2048.getStatus()).toBe('playing');
+    });
+
+    it('should be a "win" when a 2048 tile is created', () => {
+      const game2048 = new Game([
+        [0, 0, 0, 0],
+        [0, 0, 1024, 1024],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+      ]);
+
+      game2048.start();
+      game2048.moveRight();
+
+      expect(game2048.getStatus()).toBe('win');
+    });
+
+    it('should be a "lose" when no more moves are possible', () => {
+      const game2048 = new Game([
+        [128, 128, 0, 8],
+        [16, 8, 16, 32],
+        [8, 16, 32, 64],
+        [16, 32, 0, 128],
+      ]);
+
+      game2048.start();
+      game2048.moveLeft();
+
+      expect(game2048.getStatus()).toBe('lose');
+    });
+  });
+
+  describe('game score', () => {
+    it('should be updated after cells are merged horizontally', () => {
+      const game2048 = new Game([
+        [16, 0, 32, 0],
+        [32, 0, 8, 8],
+        [0, 8, 8, 0],
+        [8, 8, 0, 16],
+      ]);
+
+      game2048.start();
```

Comment:
It would be great to check the score right before the move
To ensure that the status calculation in the reason but not the initial status with a non empty cell